java/connectors/semantickernel-connectors-memory-jdbc/src/main/java/com/microsoft/semantickernel/connectors/memory/jdbc/JDBCConnector.java

```diff
@@ -76,13 +55,14 @@ public Mono<Void> createTableAsync(Connection connection) {
                                             + "metadata TEXT, "
                                             + "embedding TEXT, "
                                             + "timestamp TEXT, "
+                                            + "UNIQUE (collection, key), "
                                             + "FOREIGN KEY(collection) REFERENCES "
                                             + COLLECTIONS_TABLE_NAME
                                             + "(id)"
                                             + " )";
 
                             String createIndex =
-                                    "CREATE INDEX "
+                                    "CREATE INDEX IF NOT EXISTS "
                                             + INDEX_NAME
                                             + " ON "
                                             + TABLE_NAME
```

Comment:
For multiple calls this String concatenation will drive up allocation costs. Could this be avoided using prepared statements?